### PR TITLE
Expose whether Bluetooth is available.

### PR DESCRIPTION
**Pull request by jyasskin**
Preview at https://api.csswg.org/bikeshed/?url=https://raw.githubusercontent.com/jyasskin/web-bluetooth-1/expose-availability/index.bs#availability.

Thanks to @domenic for simplifying the overall design.

@npdoty can you see if you agree with the [privacy section](https://api.csswg.org/bikeshed/?url=https://raw.githubusercontent.com/jyasskin/web-bluetooth-1/expose-availability/index.bs#availability-fingerprint) or if there's something different I should say?

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -1684,6 +1717,100 @@ spec: html
...
+      Note: the promise is resolved <a>in parallel</a>
+      to let the UA call out to other systems
+      to determine whether Bluetooth is available.
+    </div>
+
+    <div algorithm="availabilitychanged" id="availability-changed-algorithm">
+      <p>
+        If the UA becomes able or unable to use Bluetooth,
+        for example because a radio was physically attached or detached,
+        the UA must run the following steps:
```
Comment (scheib) [inline index.bs]: Must? What about for UAs that choose to not disclose the availability?

**Comment by scheib** [comment]
Looks good to me with the exception noted regarding UAs that choose to not
disclose the availability: the wording for availabilitychanged event should
be clear that UAs may choose to not implement this.

File: index.bs [hunk 2]
```diff
@@ -1684,6 +1717,100 @@ spec: html
...
+      but websites may be able to handle it more gracefully.
+    </p>
+
+    <div class="example" id="example-getavailability">
+      <p>
+        To show Bluetooth UI only to users who have a Bluetooth adapter:
+      </p>
+      <pre highlight="js">
+      let bluetoothUI = document.querySelector('#bluetoothUI');
+      navigator.bluetooth.<a idl>getAvailability()</a>.then(available => {
```
Comment (beaufortfrancois) [inline index.bs]: I think `isAvailable` name would make it clearer this is a boolean value instead of just `available`.

File: index.bs (hunk 1 above)
Comment (jyasskin) [inline index.bs]: Good point.

Comment (jyasskin) [inline index.bs]: Done, by talking about the user configuring the UA to return a particular value.

File: index.bs (hunk 2 above)
Comment (jyasskin) [inline index.bs]: Done.

**Comment by beaufortfrancois** [comment]
👍

**Comment by jyasskin** [comment]
@msramek from Chrome's privacy team thinks this is fine, so I'm going to go ahead and submit it.

**Pull request merged**